app/code/Magento/Customer/Model/ResourceModel/CustomerRepository.php

```diff
@@ -185,6 +196,7 @@ public function __construct(
      * @throws \Magento\Framework\Exception\LocalizedException
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.NPathComplexity)
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
```

Comment:
Hi @engcom-Charlie, can we extract some logic into a separate private method instead of suppressing the warnings?
Probably a good place to extract separately is around 285-287 lines or maybe 206-209.
Thank you.